common/lib/xmodule/xmodule/modulestore/tests/test_publish.py

```diff
@@ -139,3 +157,1222 @@ def test_publish_draft_delete(self):
+            )
+            self.assertIsNotNone(
+                parent_id,
+                msg="Parent block id must be passed for draft {} item!".format(block_type)
+            )
+            self.assertIsNotNone(
+                index_in_children_list,
+                msg="Index within {} must be passed for draft {} item!".format(parent_type, block_type)
+            )
+            parent_url_regex = '({DEPRECATED_PARENT_KEY}|{PARENT_KEY})'.format(
+                DEPRECATED_PARENT_KEY=self._make_deprecated_block_key(course_key, parent_type, parent_id),
+                PARENT_KEY=self._make_block_key(course_key, parent_type, parent_id),
+            )
+            child_index_regex = '{}'.format(index_in_children_list)
+
+        # Form the checked attributes based on the block type.
+        attrs = {}
+        if block_type == 'html':
+            filename = kwargs.pop('filename', None)
+            attrs.update({'filename': filename})
+        else:
+            attrs.update({
+                'parent_url': parent_url_regex,
+                'index_in_children_list': child_index_regex
+            })
+
+        # If children exist, construct regular expressions to check them.
+        child_id_regex = None
+        child_type = None
+        child_types_ids = kwargs.pop('child_ids', None)
+        if child_types_ids:
+            # Grab the type of the first child as the type of all the children.
+            child_type = child_types_ids[0][0]
+            # Construct regex out of all the child_ids that are included.
+            child_id_regex = '|'.join([child[1] for child in child_types_ids])
+
+        return {
+            'tag': block_type,
+            'attrs': attrs,
+            'children': {
+                'tag': child_type,
+                'attrs': {'url_name': child_id_regex},
+            }
+        }
+
+    def _assertOLXBase(self, block_list, draft, published):  # pylint: disable=invalid-name
+        """
+        Check that all blocks in the list are draft blocks in the OLX format when the course is exported.
+        """
+        for block_data in block_list:
+            block_params = self.course_db.get(block_data)
+            self.assertIsNotNone(block_params)
+            (block_type, block_id) = block_data
+            if draft:
+                xml_parse_regex = self._make_xml_parse_regex(block_type, self.course_key, draft=True, **block_params)
+                self.assertOLXContent(block_type, block_id, draft=True, xml_parse=xml_parse_regex)
+            else:
+                self.assertOLXMissing(block_type, block_id, draft=True)
+            if published:
+                xml_parse_regex = self._make_xml_parse_regex(block_type, self.course_key, draft=False, **block_params)
+                self.assertOLXContent(block_type, block_id, draft=False, xml_parse=xml_parse_regex)
+            else:
+                self.assertOLXMissing(block_type, block_id, draft=False)
```

Comment:
Should this use the `bulk_assertions` tools that we wrote, so that you find out all of the errors for all of the blocks at the same time?